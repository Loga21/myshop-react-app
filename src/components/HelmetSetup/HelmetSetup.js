import React from 'react';
// importing PropTypes package
import PropTypes from 'prop-types';
// importing helmet package
import { Helmet, HelmetProvider } from 'react-helmet-async';

// Fn comp with anonymous function
// comp can have js, jsx and css (optional)
// destructuring the props
const HelmetSetup = ({ title }) => {
  // must return JSX
  return (
    <HelmetProvider>
      <Helmet>
        {/* receiving the title as prop  */}
        <title data-testid='title'>{title}</title>
      </Helmet>
    </HelmetProvider>
  );
};

// validating the prop types
HelmetSetup.propTypes = {
  title: PropTypes.string
};

export default HelmetSetup;
