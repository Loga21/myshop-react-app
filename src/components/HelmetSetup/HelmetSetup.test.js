import renderer from 'react-test-renderer';
import HelmetSetup from './HelmetSetup';

describe('HelmetSetup', () => {
  const props = {
    title: 'Title'
  };
  it('receives `title` as prop', () => {
    // test spec
    const pageTitle = renderer.create(<HelmetSetup {...props} />).toJSON();
    expect(pageTitle).toMatchSnapshot();
  });
});
