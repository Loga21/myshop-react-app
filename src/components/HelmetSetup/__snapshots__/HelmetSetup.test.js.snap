// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelmetSetup receives \`title\` as prop 1`] = `null`;

exports[`Page Title receives \`title\` prop 1`] = `null`;
